sdcm/kcl_thread.py

```diff
@@ -78,3 +81,57 @@ def _run_stress(self, loader, loader_idx, cpu_idx):
             raise
         finally:
             KclStressEvent('finish', node=loader, stress_cmd=stress_cmd, log_file_name=log_file_name)
+
+
+class CompareTablesSizesThread(DockerBasedStressThread):  # pylint: disable=too-many-instance-attributes
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._stop_event = threading.Event()
+
+    def kill(self):
+        self._stop_event.set()
+
+    def db_node_to_query(self, loader):
+        """Select DB node in the same region as loader node to query"""
+        db_nodes = [db_node for db_node in self.node_list]  # if not db_node.running_nemesis]
+        assert db_nodes, "No node to query, nemesis runs on all DB nodes!"
+        node_to_query = random.choice(db_nodes)
+        LOGGER.debug(f"Selected '{node_to_query}' to query for local nodes")
+        return node_to_query
+
+    def _run_stress(self, loader, loader_idx, cpu_idx):
+        KclStressEvent('start', node=loader, stress_cmd=self.stress_cmd)
+        try:
+            options_str = self.stress_cmd.replace('table_compare', '').strip()
+            options = dict(item.strip().split("=") for item in options_str.split(";"))
+            interval = int(options.get('interval', 20))
+            src_table = options.get('src_table')
+            dst_table = options.get('dst_table')
+
+            while not self._stop_event.is_set():
+                node: BaseNode = self.db_node_to_query(loader)
+                node.run_nodetool('flush')
+
+                src_size = node.get_cfstats(src_table)['Number of partitions (estimate)']
+                dst_size = node.get_cfstats(dst_table)['Number of partitions (estimate)']
```

Comment:
Those are estimates only, I don't think there's any guarantee that they will be equal (even if the number of rows is the same in each table).

And even if they will, we should check if the tables are 100% equal, not just size equality.

Why not use a tool such as `scylla-migrate` like I do in the CDC replication test?

I guess size equality is a start, but eventually it should be extended (maybe in a later PR).

Reply:
I was thinking on that, anyhow the issue is that I want to communicate that, i.e. kill the hydra-kcl.
We had code that did Scans inside the hydra-kcl, via alternator api, that didn't scale that nicely

I guess I can switch `scylla-migrate`, I'll need to take a close look what it doing exactly

I was aiming at adding those new "stress" command without writing a specific test code for that flow.
Maybe it was a wrong choice :\